Shared/Realm.Shared/Realm.cs

```diff
@@ -831,6 +831,16 @@ public RealmObject ObjectForPrimaryKey(string className, string id)
         /// <exception cref="System.ArgumentNullException">If you invoke this with a standalone object.</exception>
         public void Remove(RealmObject obj)
         {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            if (!this.Equals(obj.Realm))
+            {
+                throw new RealmObjectManagedByAnotherRealmException("This object is managed by another realm. You should call Remove on the other realm.");
```

Comment:
I would say so, yes. Unless you can think of a scenario where one would want to do that without any hassle. It seems to me that it's a programmer error if you try to remove something from the realm that wasn't in there.